src/syntax/read.rkt

```diff
@@ -51,29 +58,51 @@
        [(list (cons prop val) rest ...)
         (loop (dict-set* prop-dict prop val) rest)])))
   
-  (define default-repr (get-representation (dict-ref prop-dict* ':precision 'binary64)))
+  (define default-prec (dict-ref prop-dict* ':precision 'binary64))
+  (define default-repr (get-representation default-prec))
   (define var-reprs 
     (for/list ([arg args] [arg-name arg-names])
       (cons arg-name
             (if (and (list? arg) (set-member? args ':precision))
                 (get-representation (list-ref args (add1 (index-of args ':precision))))
                 default-repr))))
 
-  (define body* (desugar-program body default-repr var-reprs))
+  ;; Named fpcores need to be added to function table
+  (when func-name
+    (hash-set! (*functions*) func-name
+               (list args default-prec body)))
+
+  ;; Try props first, then identifier, else the expression itself
+  (define name
+    (or (dict-ref prop-dict* ':name #f)
+        func-name
+        body))
+
+  ;; load conversion operators for desugaring
+  (define convs (dict-ref prop-dict* ':herbie-conversions '()))
+  (generate-conversions convs)
+
+  ;; inline and desugar
+  (define inlined (inline-functions body))
+  (define body* (desugar-program inlined default-repr var-reprs))
```

Comment:
Any reason the `inline-functions` call isn't part of `desugar-program`? Would be cleaner IMO.